doc/arch/adr-006-transition-whitehall-to-publishing-api-index.md

```diff
@@ -0,0 +1,86 @@
+all sorts of documents beyond what we normally think of as "mainstream" content.
+The `govuk` index can be rebuilt from scratch in about an hour by resending the content
+from publishing API.
+
+See [ADR 006](adr-004-transition-mainstream-to-publishing-api-index.md) for more
+details about moving mainstream content.
+
+This leaves two indexes that are populated the old way: `government` and `detailed`.
+
+`detailed` contains detailed guides and `government` contains everything else
+published by Whitehall publisher.
+
+We also have a separate worker (`publishing-queue-listener`) that listens to
+`*.links` notifications from publishing API, and *updates* the old indexes
+(but doesn't add any new content).
+
+## Decision
+
+We intend to get rid of the `government` and `detailed` indexes. Rummager
+should index Whitehall Publisher documents into the `govuk` index.
+
+All search indexing code should be removed from Whitehall Publisher.
+
+The old indexing code should be removed from Rummager:
+
+- Anything in `lib/indexer`
+- The `publishing-queue-listener` worker
+
+The [infrastructure to log indexing requests to disk](https://docs.publishing.service.gov.uk/manual/rummager-traffic-replay.html) can also be removed,
+since data from the publishing API can be resent at any time.
+
+## Consequences
+
+There are over 200,000 documents published by Whitehall Publisher.
+
+Making this change will enable us to treat all documents the same way in search,
+and we can make better use of the data we have about content to improve search,
+because there will be a single indexing process that we can change easily.
+
+For example, having this data in search would allows us to return more relevant results and/or improve the user experience of using search:
+
+- The **state** of a document, based on archiving policy and content audits
+- Different ways of grouping **document types** around user need
+- More **publishing history** details, besides the `public_updated_at` value
+- Information about **who** the content is applicable to or **where** it is relevant
+- Extensions to the topic taxonomy, like what **services** or **task lists** the content belongs to
+- Entities extracted from the text of the page (or its attachments), like **people**, **organisations**, **places**, **deadlines** and **forms**.
+
+Combining all of the indexes into a single index will change the document frequency statistics, which affects how search results are scored.
+
+## Action plan
+
+### Copy over documents as-is and test
+Before making any changes to Whitehall Publisher, we should copy all documents from the `government` and `detailed` indexes to the `govuk` index.
+
+These documents will be filtered out of search results, but will have an indirect impact on scoring of other documents.
+
+We can then run the search healthcheck to understand the impact of this. At this stage we may want to reevaluate how much each format gets boosted in the search query.
```

Comment:
It might also be useful to mark all the whitehall formats as migrated and run the healthcheck again to see the impact of the scoring changes on whitehall content. I expect we'd have to add more non-mainstream content to the healthcheck first, though, because it's quite focused on common search terms.

Reply:
Good idea, I'll try this